src/main/scala/battleship/GameActor.scala

```diff
@@ -0,0 +1,164 @@
+package battleship
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Props}
+import battleship.GameActor._
+import battleship.PlayerActor.{GetNextShot, PlaceBoats}
+import battleship.game._
+
+/**
+  * Created by m06f947 on 3-8-2017.
+  */
+
+
+
+object GameActor {
+
+  case class StartGame(boardsize:Int, player1:ActorRef, player2:ActorRef, boatset: Seq[Boat])
+
+  case class ThisIsBoatSetup(placement: Set[(Boat,BoatLocation)])
+
+  case class ThisIsNextMove(x: Int, y: Int)
+
+  case object WhatIsGameState
+
+  case class GameNotStartedYet(stage: Int)
+
+  def props: Props = Props(new GameActor)
+
+}
+
+class GameActor extends Actor with ActorLogging {
+
+
+  def placeBoats(placements: Set[(Boat, BoatLocation)]): BoardState = {
+    placements.foldLeft(BoardState.empty) {
+      case (boardState, (boat, location)) => boardState.placeBoat(boat, location)
+    }
+  }
+
+  def swapCurrentPlayer(one:ActorRef,two:ActorRef,current:ActorRef) = {
+    if (one == current) two
+    else one
+  }
+
+
+
+
+override def receive: Receive = {
+  case StartGame(size, player1, player2, boatset) => {
+    player1 ! PlaceBoats(boatset, size)
+    sender() ! "Game started!"
+    context.become(gameinit1(size, player1, player2, boatset))
+  }
+  case WhatIsGameState => sender() ! GameNotStartedYet(0)
+}
+
+  def gameinit1(size:Int,player1:ActorRef,player2:ActorRef,boatset: Seq[Boat]): Receive = {
+    case ThisIsBoatSetup(placement1) if (sender() == player1) =>

```

Comment:
A bit confusing, because normally you always need parentheses in an if statement. But there you don't have a 'then' statement which you do have here with the =>.
The IDE tells me as well, should be more aware of that.